docs/interfaces/action-service.txt

```diff
@@ -1,27 +1,19 @@
 # Action Service Interface
 
-// TODO: More Exceptions?
-Exception ActionNotFound(name: string);
+Exception ActionException(message: string);
 
 
 /**
  * Executes multiple actions in the conext of the user given by the user_id.
+ * All actions are successful if no exception is thrown, else no action was
+ * processed.
  *
- * @throws ActionNotFound
+ *
+ * @throws ActionException
  */
-execute(actions: Action[], user_id: number): ExecuteResult;
+handle_request (payload: Action[], user_id: number): void
```

Comment:
There will be actions, that produces some results, like creating a model (the id), or some status updates for the users. E.g. on mass-actions, we sometimes uses "track-ids" to visualize in the client, which objects have been successfully changed. There must be a way to give data back to the client.

Also: The name `handle_request` is misleading. This has nothing to do with requests in any case, but wich actions, that need to be performed.

Reply:
Hm. I can send back something like a list of results like `[None, None, {"some":"result"}, None]", ok?

"handle_request" seems correct to me because it is about handling a request to the action service by this handler (and its bunch of action components behind it). It is not just performing them (because if the user has no permission, nothing will be performed).

Reply:
Ok